File: app/Models/Notice.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Notice extends Model
{
    protected $fillable = [
        'school_id',
        'title',
        'description',
        'notice_date',
        'target_type',
        'class_id',
        'section_id',
        'attachment',
        'created_by',
        'status',
        'is_delete'
    ];

    protected $appends = ['attachment_url'];

    public function getAttachmentUrlAttribute(): ?string
    {
        return $this->attachment ? url('/storage/' . $this->attachment . '?t=' . ($this->updated_at ? $this->updated_at->timestamp : time())) : null;
    }
    public function school(): BelongsTo
    {
        return $this->belongsTo(School::class);
    }

    public function class(): BelongsTo
    {
        return $this->belongsTo(ClassModel::class, 'class_id');
    }

    public function section(): BelongsTo
    {
        return $this->belongsTo(Section::class);
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}

File: lib/data/models/notice.dart
class NoticeModel {
  final int id;
  final String title;
  final String description;
  final String noticeDate; // YYYY-MM-DD
  final String targetType; // Entire School, Class Wise, Section Wise
  final String? attachment;
  final CreatorModel? creator;

  NoticeModel({
    required this.id,
    required this.title,
    required this.description,
    required this.noticeDate,
    required this.targetType,
    this.attachment,
    this.creator,
  });

  factory NoticeModel.fromJson(Map<String, dynamic> json) {
    return NoticeModel(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      noticeDate: json['notice_date'] ?? '',
      targetType: json['target_type'] ?? 'Entire School',
      attachment: json['attachment'] ?? json['file_path'],
      creator: json['creator'] != null ? CreatorModel.fromJson(json['creator']) : null,
    );
  }
}

class CreatorModel {
  final int id;
  final String name;

  CreatorModel({required this.id, required this.name});

  factory CreatorModel.fromJson(Map<String, dynamic> json) {
    return CreatorModel(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}
